# Basic Programming
- Hello World
- Two Strings
- char sum
- Recursive Sum
- Profile picture

In [1]:
# Kirti is new to programming and she doesn't know how to code. Can you write a code, which would print "Hello Kirti" on the screen.

print('Hello Kirti')

Hello Kirti


# Given two strings of equal length, you have to tell whether they both strings are identical.

Two strings S1 and S2 are said to be identical, if any of the permutation of string S1 is equal to the string S2. See Sample explanation for more details.

Input :

First line, contains an intger 'T' denoting no. of test cases.
Each test consists of a single line, containing two space separated strings S1 and S2 of equal length.
Output:

For each test case, if any of the permutation of string S1 is equal to the string S2 print YES else print NO.
Constraints:

1<= T <=100
1<= |S1| = |S2| <= 10^5
String is made up of lower case letters only.
Note : Use Hashing Concept Only . Try to do it in O(string length) .

SAMPLE INPUT 
3
sumit mitsu
ambuj jumba
abhi hibb
SAMPLE OUTPUT 
YES
YES
NO

In [3]:
def string():
    t=int(input())
    for i in range(t):
        m=input()
        n=1
        x,y=m.split()
        
        for j in x:
            if x.count(j)!=y.count(j):
                n=0
                break
        if n==1:
            print("YES")
        elif n==0:
            print("NO")
string()
    

3
sumit mitsu
YES
ambuj jumba
YES
abhi hibb
NO


# You have been given an array A of size N consisting of positive integers. You need to find and print the product of all the number in this array Modulo.

Input Format:
The first line contains a single integer N denoting the size of the array. The next line contains N space separated integers denoting the elements of the array


In [2]:
def product(n,s):
    a=1
    for i in range(n):
        a=(a*int(s[i]))%((10**9)+7)
    print(a)
    return
n=int(input())
s=input().split()
product(n,s)        
            

5
1 2 3 4 5
120


#Roy wants to change his profile picture on Facebook. Now Facebook has some restriction over the dimension of picture that we can upload.
Minimum dimension of the picture can be L x L, where L is the length of the side of square.

Now Roy has N photos of various dimensions.
Dimension of a photo is denoted as W x H 
where W - width of the photo and H - Height of the photo

When any photo is uploaded following events may occur:

[1] If any of the width or height is less than L, user is prompted to upload another one. Print "UPLOAD ANOTHER" in this case.
[2] If width and height, both are large enough and 
(a) if the photo is already square then it is accepted. Print "ACCEPTED" in this case.
(b) else user is prompted to crop it. Print "CROP IT" in this case.

(quotes are only for clarification)

Given L, N, W and H as input, print appropriate text as output.

Input:
First line contains L.
Second line contains N, number of photos.
Following N lines each contains two space separated integers W and H.

Output:
Print appropriate text for each photo in a new line.

Constraints:
1 <= L,W,H <= 10000
1 <= N <= 1000

SAMPLE INPUT 
180
3
640 480
120 300
180 180
SAMPLE OUTPUT 
CROP IT
UPLOAD ANOTHER
ACCEPTED

In [6]:
def profilepicture(length,width,height):
    if(width<length or height<length):
        print("UPLOAD ANOTHER")
    elif(width==height):
        print("ACCEPTED")
    else:
        print("CROP IT")
    return 
    
 
 
length=int(input())
n=int(input())
for i in range(n):
    s=input().split()
    width=int(s[0])
    height=int(s[1])
    profilepicture(length,width,height)
    

180
3
640 480
CROP IT
120 300
UPLOAD ANOTHER
180 180
ACCEPTED


Little Bear has received a home assignment to find the sum of all digits in a number N. Following his affinity towards single digit number, he intends to repeatedly compute the sum of all digits until the sum itself becomes a single digit number.

Can you write a program to compute the final single-digit sum?

As the number N is very big, it is given in the following run length encoded format - N is represented as a sequence of M blocks, where each block i (0 ≤ i < M) is represented by two integers - (len[i], d[i]). This implies that the digit d[i] occurs len[i] number of times.

For example, {(2,1), (1,2), (2,9)} represents the number 11299.

Input:
The first line contains a single integer T, denoting the number of test cases.
Each test case starts with a single integer M, which is the number of following run-length encoded blocks.
Each of the next M lines contains two space separated integers: len[i] and d[i].

Output:
Print the single-digit sum for each of the T test cases in a new line.

Constraints:
0 ≤ d[i] ≤ 9
1 ≤ len[i] ≤ 1015
1 ≤ M ≤ 104
1 ≤ T ≤ 10

SAMPLE INPUT 
3
3
2 1
1 2
2 9
1
8 1
3
2 5
1 4
1 5

SAMPLE OUTPUT 
4
8
1

In [7]:
def recursivesum():
    test=int(input())
    total=0
    for j in range(test):
        d= input().split()
        a=int(d[0])
        b=int(d[1])
        total=total+a*b
    if total<=9:
        return total
    else:
        while total>9:
            check=list(str(total))
            check=[int(x) for x in check]
            total=sum(check)
        return total
n=int(input())
for a in range(n):
    print(recursivesum())
    

3
3
2 1
1 2
2 9
4
1 
8 1
8
3
2 5
1 4
1 5
1


# Char sum
Consider All lowercase Alphabets of the English language. Here we consider each alphabet from a to z to have a certain weight. The weight of the alphabet a is considered to be 1, b to be 2, c to be 3 and so on until z has a weight of 26. In short, the weight of the alphabet a is 1, and the weight of all other alphabets is the weight of its previous alphabet + 1.

Now, you have been given a String S consisting of lowercase English characters. You need to find the summation of weight of each character in this String.

For example, Consider the String aba

Here, the first character a has a weight of 1, the second character b has 2 and the third character a again has a weight of 1. So the summation here is equal to : 1+2+1=4

Input Format:

The first and only line of input contains the String S.

Output Format:

Print the required answer on a single line

Constraints:

1≤|S|≤100

SAMPLE INPUT 
aba
SAMPLE OUTPUT 
4

In [8]:
s=input()
def charsum(s):
    c=0
    for i in s:
        c+=ord(i)-96
    print(c)
charsum(s)

aba
4


In [5]:
def maze(n):
    x=0
    y=0
    for i in n:
        if i=='L':
            x=x-1
        if i=='R':
            x=x+1
        if i=='U':
            y=y+1
        if i=='D':
            y=y-1
        print(x,y)
maze('LLRDDR')

-1 0
-2 0
-1 0
-1 -1
-1 -2
0 -2


In [ ]:

def lift(n):
    a=0
    b=7
    for i in range(n):
        m=int(input())  # user standing position
        c = abs(a-m)    # lift - user
        d = abs(b-m)# lift - user
        if i<=7:
            print('out of range')
        if (c <= d):
            print("A")
            a=m
        else:
            print('B')
            b=m
    return
n=int(input()) # test cases
lift(n)

2
8
out of range
B


In [ ]:
def seating(n):
    for i in range(0,n+1):
        seat=int(input())
        m=seat-1//12
        if seat%3==2:
            a="MS"
        if seat%6<2:
            a="WS"
        else:
            a="AS"
        print(24*m+13-seat,a)
    
n=int(input())
seating(n)

In [3]:
def life(n):
    while True:
        if n==42:
            break
        print(n)

n=int(input())
life(n)

42
